// office-ui-fabric-react/lib-amd/components/Stack/examples/Stack.Horizontal.Reversed.Example.js
define(["require", "exports", "tslib", "react", "office-ui-fabric-react"], function (require, exports, tslib_1, React, office_ui_fabric_react_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    // Styles definition
    var stackStyles = {
        root: {
            background: office_ui_fabric_react_1.DefaultPalette.themeTertiary
        }
    };
    var stackItemStyles = {
        root: {
            background: office_ui_fabric_react_1.DefaultPalette.themePrimary,
            color: office_ui_fabric_react_1.DefaultPalette.white,
            padding: 5
        }
    };
    var itemAlignmentsStackStyles = {
        root: {
            background: office_ui_fabric_react_1.DefaultPalette.themeTertiary,
            height: 100
        }
    };
    // Tokens definition
    var containerStackTokens = { childrenGap: 5 };
    var horizontalGapStackTokens = {
        childrenGap: 10,
        padding: 10
    };
    var itemAlignmentsStackTokens = {
        childrenGap: 5,
        padding: 10
    };
    var clickableStackTokens = {
        padding: 10
    };
    var HorizontalStackReversedExample = /** @class */ (function (_super) {
        tslib_1.__extends(HorizontalStackReversedExample, _super);
        function HorizontalStackReversedExample() {
            var _this = _super !== null && _super.apply(this, arguments) || this;
            _this._onClick = function () {
                alert('Clicked Stack');
            };
            return _this;
        }
        HorizontalStackReversedExample.prototype.render = function () {
            return (React.createElement(office_ui_fabric_react_1.Stack, { tokens: containerStackTokens },
                React.createElement("span", null, "Default horizontal stack"),
                React.createElement(office_ui_fabric_react_1.Stack, { horizontal: true, reversed: true, disableShrink: true, styles: stackStyles },
                    React.createElement("span", null, "Item One"),
                    React.createElement("span", null, "Item Two"),
                    React.createElement("span", null, "Item Three")),
                React.createElement("span", null, "Horizontal gap between items"),
                React.createElement(office_ui_fabric_react_1.Stack, { horizontal: true, reversed: true, disableShrink: true, styles: stackStyles, tokens: horizontalGapStackTokens },
                    React.createElement("span", null, "Item One"),
                    React.createElement("span", null, "Item Two"),
                    React.createElement("span", null, "Item Three")),
                React.createElement("span", null, "Item alignments"),
                React.createElement(office_ui_fabric_react_1.Stack, { horizontal: true, reversed: true, disableShrink: true, styles: itemAlignmentsStackStyles, tokens: itemAlignmentsStackTokens },
                    React.createElement(office_ui_fabric_react_1.Stack.Item, { align: "auto", styles: stackItemStyles },
                        React.createElement("span", null, "Auto-aligned item")),
                    React.createElement(office_ui_fabric_react_1.Stack.Item, { align: "stretch", styles: stackItemStyles },
                        React.createElement("span", null, "Stretch-aligned item")),
                    React.createElement(office_ui_fabric_react_1.Stack.Item, { align: "baseline", styles: stackItemStyles },
                        React.createElement("span", null, "Baseline-aligned item")),
                    React.createElement(office_ui_fabric_react_1.Stack.Item, { align: "start", styles: stackItemStyles },
                        React.createElement("span", null, "Start-aligned item")),
                    React.createElement(office_ui_fabric_react_1.Stack.Item, { align: "center", styles: stackItemStyles },
                        React.createElement("span", null, "Center-aligned item")),
                    React.createElement(office_ui_fabric_react_1.Stack.Item, { align: "end", styles: stackItemStyles },
                        React.createElement("span", null, "End-aligned item"))),
                React.createElement("span", null, "Clickable stack"),
                React.createElement(office_ui_fabric_react_1.Stack, { horizontal: true, onClick: this._onClick, styles: stackStyles, tokens: clickableStackTokens },
                    React.createElement("span", null, "Click inside this box"))));
        };
        return HorizontalStackReversedExample;
    }(React.Component));
    exports.HorizontalStackReversedExample = HorizontalStackReversedExample;
});
